src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,460 @@
+                watcher.Filters.Add("*.dll");
+
+                watcher.Filters.Clear();
+                Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters[0]);
+                Assert.Equal(0, watcher.Filters.Count);
+                Assert.Empty(watcher.Filters);
+                Assert.NotNull(watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void InvalidOperationsOnFilters()
+        {
+            var watcher = new FileSystemWatcher();
+            watcher.Filters.Add("*.pdb");
+            watcher.Filters.Add("*.dll");
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters.Insert(4, "*"));            
+            watcher.Filters.Clear();
+            Assert.Throws<ArgumentOutOfRangeException>(() => watcher.Filters[0]);
+        }
+
+        [Fact]
+        public void SetAndGetFilterProperty()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path, "*.pdb");
+                watcher.Filters.Add("foo");
+                Assert.Equal(2, watcher.Filters.Count);
+                Assert.Equal(new string[] { "*.pdb", "foo" }, watcher.Filters);
+
+                watcher.Filter = "*.doc";
+                Assert.Equal(1, watcher.Filters.Count);
+                Assert.Equal("*.doc", watcher.Filter);
+                Assert.Equal("*.doc", watcher.Filters[0]);
+                Assert.Equal(new string[] { "*.doc" }, watcher.Filters);
+
+                watcher.Filters.Clear();
+                Assert.Equal("*", watcher.Filter);
+            }
+        }
+
+        [Fact]
+        public void SetAndGetFiltersProperty()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path, "*.pdb");
+                watcher.Filters.Add("foo");
+                Assert.Equal(new string[] { "*.pdb", "foo" }, watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Delete_MultipleFilters()
+        {
+            // Check delete events against multiple filters
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            FileInfo fileOne = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
+            FileInfo fileTwo = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
+            fileOne.Create().Dispose();
+            fileTwo.Create().Dispose();
+
+            using (var watcher = new FileSystemWatcher(directory.FullName))
+            {
+                watcher.Filters.Add(fileOne.Name);
+                watcher.Filters.Add(fileTwo.Name);
+
+                // using expectedPaths instead of expectedPath because otherwise in this call to ExpectEvent first lambda event handler (added during the last call) will fail.
```

Comment:
You should fix the helper, not degrade the test to deal with a bug.
https://stackoverflow.com/questions/859936/unhooking-events-with-lambdas-in-c-sharp